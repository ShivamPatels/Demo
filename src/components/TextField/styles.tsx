import { StyleSheet } from "react-native";
import colors from "../../utils/colors";

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: "flex-end",
        padding: 5,
    },
    inputStyle: {
        height: 48,
        flex: 1,
        fontSize: 15,
        fontWeight: "normal",
        padding: 0,
        color: colors.FIELD_TXT_COLOR,
    },
});

export default styles;
