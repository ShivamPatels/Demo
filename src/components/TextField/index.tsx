import React, { useState } from "react";
import {
    View,
    TextInput,
    TouchableOpacity,
    StyleProp,
    TextStyle,
    KeyboardType,
    Text,
} from "react-native";
import colors from "../../utils/colors";

import styles from "./styles";

enum AutoCapitalize {
    NONE = "none",
    SENTENCES = "sentences",
    WORDS = "words",
    CHARACTERS = "characters",
    UNDEFINED = "",
}

interface Props {
    mainStyle?: StyleProp<TextStyle>;
    title?: string;
    editable?: boolean;
    onChangeText?: (text: string) => void;
    placeholder?: string;
    keyboardType?: KeyboardType;
    secureTextEntry?: boolean;
    autoCapitalize?: AutoCapitalize;
    value?: string;
    onBlur?: () => {};
    maxLength?: number;
    errorMessage?: string;
    rightIcon?: any;
    isHideShow?: boolean;
    multiline?: boolean;
}

const TextField: React.FC<Props> = ({
    mainStyle,
    title,
    editable,
    onChangeText,
    placeholder,
    keyboardType,
    secureTextEntry,
    autoCapitalize,
    value,
    onBlur,
    maxLength,
    errorMessage,
    rightIcon,
    isHideShow,
    multiline,
}) => {
    const [isShow, setIsShow] = useState(false);
    const onPressHideShow = () => {
        setIsShow(!isShow);
    };

    return (
        <View style={[mainStyle]}>
            {title && (
                <Text
                    style={{
                        fontWeight: "400",
                        marginVertical: 10,
                        color: colors.BLACK,
                    }}
                >
                    {title}
                </Text>
            )}

            <View
                style={{
                    height: 48,
                    borderRadius: 10,
                    paddingHorizontal: 10,
                    borderWidth: 1,
                    borderColor:
                        !!errorMessage && errorMessage !== ""
                            ? colors.ERROR_FIELD
                            : colors.TRANSPARENT,
                    backgroundColor: colors.FIELD_BG,
                }}
            >
                <View style={{ flexDirection: "row", alignItems: "center" }}>
                    <TextInput
                        style={styles.inputStyle}
                        editable={editable}
                        placeholderTextColor={colors.PLACEHOLDER}
                        onChangeText={onChangeText}
                        placeholder={placeholder}
                        keyboardType={keyboardType}
                        secureTextEntry={isHideShow ? !isShow : secureTextEntry}
                        autoCapitalize={autoCapitalize || "none"}
                        value={value}
                        autoCorrect={false}
                        onBlur={onBlur}
                        maxLength={maxLength}
                        multiline={multiline}
                    />
                </View>
            </View>

            <View>
                {!!errorMessage && errorMessage !== "" && (
                    <Text
                        style={{
                            marginTop: 5,
                            color: colors.ERROR_FIELD,
                        }}
                    >
                        {errorMessage}
                    </Text>
                )}
            </View>
        </View>
    );
};

TextField.defaultProps = {
    multiline: false,
};

export default TextField;
