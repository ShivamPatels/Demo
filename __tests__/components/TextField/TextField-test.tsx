/**
 * @format
 */

import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import TextField from "../../../src/components/TextField";

it("renders correctly", () => {
    const tree = renderer.create(<TextField />).toJSON();
    expect(tree).toMatchSnapshot();
});
