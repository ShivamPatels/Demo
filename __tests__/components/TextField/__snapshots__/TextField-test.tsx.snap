// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#F2F2F7",
        "borderColor": "transparent",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 48,
        "paddingHorizontal": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        multiline={false}
        placeholderTextColor="rgba(60, 60, 67, 0.3)"
        style={
          Object {
            "color": "#060518",
            "flex": 1,
            "fontSize": 15,
            "fontWeight": "normal",
            "height": 48,
            "padding": 0,
          }
        }
      />
    </View>
  </View>
  <View />
</View>
`;
